### PR TITLE
feat: Custom fireclawl API URL

### PR DESCRIPTION
**Pull request by lvjiajjjlzy**
# Description

Custom fireclawl API URL

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [x] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by crazywoola** [comment]
Please do not modify the PR template :)

**Comment by lvjiajjjlzy** [comment]
> Please do not modify the PR template :)

It's changed back

**Review by crazywoola — CHANGES_REQUESTED** [review]
I am getting this error: 
```json
{
    "code": "invalid_param",
    "message": "FirecrawlApp.__init__() got an unexpected keyword argument 'api_url'",
    "status": 400
}
```

**Comment by lvjiajjjlzy** [comment]
> I am getting this error:
> 
> ```json
> {
>     "code": "invalid_param",
>     "message": "FirecrawlApp.__init__() got an unexpected keyword argument 'api_url'",
>     "status": 400
> }
> ```

I think it's because the dependency version of the Python library is too low. I just changed it.
Update pyproject.toml firecraw to 0.0.16

File: api/pyproject.toml [hunk 1]
```diff
@@ -172,7 +172,7 @@ pydantic = "~2.7.4"
 pydantic_extra_types = "~2.8.1"
 pydantic-settings = "~2.3.3"
 pgvecto-rs = "0.1.4"
-firecrawl-py = "0.0.5"
+firecrawl-py = "0.0.16"
```
Comment (Yeuoly) [inline api/pyproject.toml:175]: pls run dev/sync-poetry to sync lock file

**Review by Yeuoly — CHANGES_REQUESTED** [review]

File: api/pyproject.toml (hunk 1 above)
Comment (lvjiajjjlzy) [reply api/pyproject.toml:175]: > pls run dev/sync-poetry to sync lock file

done

File: api/poetry.lock [hunk 2]
```diff
@@ -2089,8 +2089,8 @@ description = "Python SDK for Firecrawl API"
 optional = false
 python-versions = "*"
 files = [
-    {file = "firecrawl-py-0.0.5.tar.gz", hash = "sha256:3d1cc30b7d86c12aa06e6434ebb526072cd70ab9a0c8b145008efe044a1cd09c"},
```
Comment (Yeuoly) [inline api/poetry.lock:2092]: In line 2087, version of firecrawl is still 0.0.5, is there a mistake?

**Review by Yeuoly — CHANGES_REQUESTED** [review]

**Comment by lvjiajjjlzy** [comment]
this



---- Replied Message ----
| From | ***@***.***> |
| Date | 06/24/2024 16:22 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [langgenius/dify] feat: Custom fireclawl API URL (PR #5519) |

@Yeuoly requested changes on this pull request.

In api/poetry.lock:

> @@ -2089,8 +2089,8 @@ description = "Python SDK for Firecrawl API"
 optional = false
 python-versions = "*"
 files = [
-    {file = "firecrawl-py-0.0.5.tar.gz", hash = "sha256:3d1cc30b7d86c12aa06e6434ebb526072cd70ab9a0c8b145008efe044a1cd09c"},


In line 2087, version of firecrawl is still 0.0.5, is there a mistake?

**Comment by Yeuoly** [comment]
> this
> ---- Replied Message ---- | From | ***@***.***> | | Date | 06/24/2024 16:22 | | To | ***@***.***> | | Cc | ***@***.***>***@***.***> | | Subject | Re: [langgenius/dify] feat: Custom fireclawl API URL (PR #5519) | @Yeuoly requested changes on this pull request. In api/poetry.lock:
> @@ -2089,8 +2089,8 @@ description = "Python SDK for Firecrawl API"
> optional = false python-versions = "*" files = [ - {file = "firecrawl-py-0.0.5.tar.gz", hash = "sha256:3d1cc30b7d86c12aa06e6434ebb526072cd70ab9a0c8b145008efe044a1cd09c"}, In line 2087, version of firecrawl is still 0.0.5, is there a mistake? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

pls take a look at CI errors, it looks like lock file has not been synced yet.

**Comment by lvjiajjjlzy** [comment]
Alright, I'll make the changes later.



---- Replied Message ----
| From | ***@***.***> |
| Date | 06/24/2024 16:35 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [langgenius/dify] feat: Custom fireclawl API URL (PR #5519) |

this
---- Replied Message ---- | From | @.> | | Date | 06/24/2024 16:22 | | To | @.> | | Cc | @.>@.> | | Subject | Re: [langgenius/dify] feat: Custom fireclawl API URL (PR #5519) | @Yeuoly requested changes on this pull request. In api/poetry.lock:
@@ -2089,8 +2089,8 @@ description = "Python SDK for Firecrawl API"
optional = false python-versions = "" files = [ - {file = "firecrawl-py-0.0.5.tar.gz", hash = "sha256:3d1cc30b7d86c12aa06e6434ebb526072cd70ab9a0c8b145008efe044a1cd09c"}, In line 2087, version of firecrawl is still 0.0.5, is there a mistake? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: @.**>

pls take a look at CI errors, it looks like lock file has not been synced yet.

**Comment by Yeuoly** [comment]
emmmm, seems that the error still exists

**Pull request closed without merging**